### PR TITLE
[SDK-4043] `transition:` method refactoring

### PR DESCRIPTION
**Pull request by maratal**
Partially solves https://github.com/ably/ably-cocoa/issues/1858
This should be merged after https://github.com/ably/ably-cocoa/pull/1714

Summary:
Now it's a single method for transition with more clear name and it passes `resume` through metadata directly rather than in two steps (metadata->state change->method). Read commit messages for more.

### CODE REVIEW

**Review by sacOO7 — CHANGES_REQUESTED** [review]

File: Source/ARTRealtime.m [hunk 1]
```diff
@@ -910,7 +900,7 @@ - (void)onConnected:(ARTProtocolMessage *)message {
             }
             ARTConnectionStateChangeMetadata *const metadata = [[ARTConnectionStateChangeMetadata alloc] initWithErrorInfo:message.error];
             metadata.resumed = resumed;  // RTN19a
```
[the remark this answers is not in the document]
Comment (maratal) [reply Source/ARTRealtime.m:902]: I got your point, I can just add one more parameter alongside metadata. But metadata is just a set of parameters. In this regard I would just rename it to `ARTConnectionStateChangeParams` or `ARTConnectionStateChangeArgs` as methods with a lot of parameters sometimes do, it's a common pattern. WDYT? @sacOO7 @lawrence-forooghian

[the remark this answers is not in the document]
Comment (sacOO7) [reply Source/ARTRealtime.m:902]: Yeah, I feel renaming is a good idea 👍

[the remark this answers is not in the document]
Comment (maratal) [reply Source/ARTRealtime.m]: https://github.com/ably/ably-cocoa/pull/1863/commits/d75c5c1f0de9ef6f332a1d8a6c69484b02fa69ac

**Review by sacOO7 — APPROVED** [review]
LGTM

**Review by lawrence-forooghian — APPROVED** [review]

**Pull request merged**